runtime/bcutil/ConstantPoolMap.hpp

```diff
@@ -348,6 +348,9 @@ class ConstantPoolMap
 		PUT_FIELD = SPLIT1,
 		GET_FIELD = SPLIT1,
 		NEW = SPLIT1,
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		DEFAULT_VALUE = SPLIT1,
```

Comment:
Sorry @eric98zhang I missed your response.  

Both `new` and `defaultvalue` are able to point to the same constantpool entry.  Unless there's an update to the JVMS to indicate otherwise, the verifier won't track which bytecode uses which cp entry - only that the data is consistent.  In the case of `new` and `defaultvalue`, both  bytecodes use the CONSTANT_Class cpEntries so there's nothing else for the verifier to check.

This is the kind of issue that needs to be handled at resolution or in the bytecodes.  In OpenJ9, each cpEntry is two slots and a bytecode like `new` relies on the values in those slots to determine whether its resolved or not.  

Once the cpEntry is resolved, all users share that resolved status so we need to either not resolve a cpEntry if its used by both new and defaultvalue (slow), or find another way to encode the data so that both bytecodes have unique resolution states, or to add a check in the bytecode to validate the resolved value is of the right kind (class vs value).

OpenJ9's design has typically been to aim for finding the right data encoding so that all the cost is at resolution time, not execution time.  

@tajila pointed out that `new` already has a check though for whether the resolved J9Class is `abstract` or `interface` so adding a new case to this for `value` seems reasonable and no more expensive then current code.